server/command.go

```diff
@@ -236,11 +236,16 @@ func (p *Plugin) executeLinkCommand(args *model.CommandArgs, parameters []string
 		return p.cmdError(args.UserId, args.ChannelId, "Error occurred while saving the subscription")
 	}
 
-	if err := p.store.CommitTx(tx); err != nil {
+	if err = p.store.CommitTx(tx); err != nil {
 		p.API.LogError("Unable to commit database transaction", "error", err.Error())
 		return p.cmdError(args.UserId, args.ChannelId, "Something went wrong")
 	}
 
+	err = p.store.UpdateSubscriptionSyncNeeded(channelsSubscription.ID, false)
+	if err != nil {
+		p.API.LogWarn("Unable to mark the subscription as no sync needed", "subscriptionID", channelsSubscription.ID, "error", err.Error())
+	}
```

Comment:
I would have expected this to be set to `false` in the call to `SaveChannelSubscription`. Is there a reason we don't do that?

Reply:
This was in fact something that we can remove all together, the whole function from the store even.